chain-tx-enclave/enclave-utils/src/crypto/key_exchange.rs

```diff
@@ -0,0 +1,123 @@
+use std::io::Write;
```

Comment:
when is this custom key exchange used?

Reply:
https://github.com/ndokmai/rust-sgx-remote-attestation/blob/7e2c26930d4a87aa040b3e1d5602c61fcd5145ee/ra-sp/src/context.rs#L142

Reply:
it's copying that example with a modified Sigma protocol: https://github.com/intel/sgx-ra-sample I guess?
Which component will need a communication over Sigma? Here, the end goal is to have an attested TLS client<->server channel, so one needs HTTPS to IAS, processing its payloads, AESM comms + verification of quotes (checking CMAC, SHA256(nonce||quote)... )